content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/future-work.md

```diff
@@ -0,0 +1,55 @@
+---
+linktitle: Future Work
+title: Future Work
+description: Possible future improvements to the terrain system.
+weight: 500
+toc: true
+---
+
+There are a number of improvements that could potentially be made to the terrain system for anyone interested in taking the system further.
+
+## Large-scale improvements
+
+This category of improvements requires architecting new solutions to provide extensive new capabilities.
+
+### 3D geometry support
+
+Instead of a heightfield-based approach, the APIs and renderer could be extended to use a full 3D approach, whether it's through voxels, displacement maps, mesh ingestion, raytracing, or some other technology. This would enable the system to support caves, arches, cliff geometry, and other vertical or overlapping terrain features that simply can't be supported through a pure heightfield approach.
+
+The API extensions would likely consist of redefining the current APIs to provide the _highest elevation_ terrain data that appears below the given query input positions, and then adding more APIs to provide _all_ the terrain data that exists at or below the given query positions.
+
+### GPU calculation support
+
+The terrain system is built on top of the gradient components, which perform all of their calculations CPU-side. However, the types of calculations being performed are ideal for GPU-side calculation. An overhaul or replacement of the gradient system that supports GPU-side calculations could provide orders of magnitude improvements to the terrain system. The primary caveats are that the data is still needed CPU-side to feed to physics, generalized raycasts, and other systems, and that it should be possible to still use the terrain system on devices with limited or no GPU capabilities (low-end phones, headless servers, etc).
```

Comment:
For various reasons, I suggest using a simpler adjective, rather than orders of magnitude. The primary reason is that it's long and I think breaks the reading flow. 
```suggestion
The terrain system is built on top of the gradient components, which perform all of their calculations on the CPU-side. However, these calculations can be computed more optimally on the GPU-side. An overhaul or replacement of the gradient system that supports GPU-side calculations can provide significant improvements to the terrain system's performance. The primary caveats are that the data is still needed on the CPU-side to feed to physics, generalized raycasts, and other systems, and that it should be possible to still use the terrain system on devices with limited or no GPU capabilities (low-end phones, headless servers, etc).
```

Reply:
How about:
```suggestion
The terrain system is built on top of the gradient components, which perform all of their calculations on the CPU. However, a GPU can perform bulk computations like these far more optimally. An overhaul or replacement of the gradient system that supports GPU-side calculations can improve the terrain system's performance by orders of magnitude. The primary caveats are that the data is still needed on the CPU to feed to physics, generalized raycasts, and other systems, and that it should be possible to still use the terrain system on devices with limited or no GPU capabilities (low-end phones, headless servers, etc).
```

The phrase "orders of magnitude" is more specific and objective than "significant" so I'd like to preserve it.

Reply:
I wasn't sure if it was figurative or not, so in that case, agree to keep orders of magnitude, and your edit looks good.